frb_dart/lib/src/generalized_frb_rust_binding/_io.dart

```diff
@@ -5,6 +5,17 @@ import 'package:flutter_rust_bridge/src/generalized_uint8list/generalized_uint8l
 import 'package:flutter_rust_bridge/src/platform_types/_io.dart';
 import 'package:flutter_rust_bridge/src/platform_types/platform_types.dart';
 
+final class ShutdownWatcher implements ffi.Finalizable {
+  static var _finalizer;
+
+  ShutdownWatcher(ffi.Pointer<ffi.NativeFinalizerFunction> callback) {
+    _finalizer = ffi.NativeFinalizer(callback);
```

Comment:
nit: maybe we can follow the pattern shown in https://api.flutter.dev/flutter/dart-ffi/NativeFinalizer-class.html to handle this, e.g. demo there uses

```
  static final _finalizer =
      NativeFinalizer(_nativeDatabaseBindings.closeDatabaseAddress.cast());
```

etc

Reply:
We dont have `ExternalLibrary` available in a static context - it is provided via `GeneralizedFrbRustBinding` constructor, so can't make it exactly like this. But having `_ShutdownWatcher` as a global variable or a static field should have the same guarantees.

Reply:
I see. Could you please also make the reasoning in comments - then it would be more clear why everything is like that for future readers